Dockerfile

```diff
@@ -0,0 +1,21 @@
+# certs
+FROM alpine:3.13 as certs
+RUN apk add -U --no-cache ca-certificates
+
+# builder
+FROM golang:1.17 as build
+ARG cmd
+WORKDIR /app
+# Download dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+# Copy over the rest of the code
+COPY . .
+# Build the specified cmd
+RUN GOOS=linux CGO_ENABLED=0 GOGC=off GOARCH=amd64 go build -o "./bin/${cmd}" "./cmd/${cmd}"
```

Comment:
You might do better for caching if you directly specify each of the cmds

Reply:
I might be missing your point. Each service builds separately at the moment ([e.g. api service](https://github.com/alexdunne/gs-onboarding/blob/GSIS-1382-dockerise-services/docker-compose.yml#L23)) so I'm not sure where the improved caching would come from?